### DISCUSSION TITLE
[Kavita] Search within ePubs

### DISCUSSION THREAD

**Discussion by DieselTech** [Kareadita/Kavita#2535] [Ideas]
### Idea Description

Posted by Samuel Halle
· Jun 2022
· 44 Upvotes
https://feats.kavitareader.com/posts/150/search-within-epubs

Description

I have a lot of recipes books and it would be awesome to have the ability to do a text search for some words, either in the Table of content or the full content of the book.

### Duration of Using Kavita

42

### Idea Category

Feature Enhancement

### Agreement

- [X] I agree that this is solely for submitting ideas, and I will search for existing ideas before posting.

**Comment by valzevul** [comment]
This would be really nice to have. 

I have a similar use-case: lots of ePubs / OCR'ed PDFs with recipes, and I'd love to have a way to do a full-text search across all books at once.

For example, I type "tarragon" in the search field and get a list of all books each with a short excerpt featuring a few sentences with the searched word.

Some examples of apps supporting this are calibre (not calibre-web though), Adobe Reader (as a standalone app), and paperless-ngx.

**Comment by meyou69** [comment]
Absolutely agree. I'm genuinely surprised that this hasn't already been implemented yet. Feels like such a huge missed opportunity.

It would be a great quality of life enhancement

**Comment by HamletDuFromage** [comment]
Would be awesome, strong +1
